VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/services/ProviderService.java

```diff
@@ -28,4 +29,30 @@ public Provider persistProvider(Provider provider) throws SQLException {
         provider.setProviderId(providerId);
         return provider;
     }
+
+    public List<Provider> getProvider(String id) throws Exception{
+        long num_id =Long.parseLong(id);
+        try {
+            return providerDao.getProvider(num_id);
+        }
+        catch(Exception e){
+            System.out.println(e);
+            throw new Exception();
+        }
+    }
+
+    /**
+     *
+     * @return List of providers, empty list if no providers
+     * @throws Exception if cascaded from DAO.
+     */
+    public List<Provider> getProviders() throws Exception{
+        try {
+            return providerDao.getProviders();
+        }
+        catch(Exception e){
+            System.out.println(e);
+            throw new Exception();
+        }
```

Comment:
Its good to have exception handling, but currently, this function only relays calls to the DAO. Hence, exception handling is not needed at this instance. Even if we keep this, printing to "System.out" might not be needed.

Reply:
He is right in handling the exception here. Please refer the providerDao's getProviders() method it throws an SQL exception. However, instantiating a new exception object is not needed here. We already have the exception from the dao, we can just print the stack trace with a message. We don't need to have both a throws clause and exception handling at the same time. 
Also, Cristian - refer to the comment I made in the dao class regarding exception. This needs to be in sync with that.